src/aiq/front_ends/mcp/mcp_front_end_plugin.py

```diff
@@ -0,0 +1,408 @@
+                if hasattr(function_config, "topic") and function_config.topic:
+                    function_description = function_config.topic
+                # Try to get description from config
+                elif hasattr(function_config, "description") and function_config.description:
+                    function_description = function_config.description
+                # Try to get description directly from the function
+                elif hasattr(function, "description") and function.description:
+                    function_description = function.description
+                # Fall back to function docstring
+                elif hasattr(function, "__doc__") and function.__doc__:
+                    # Extract the first paragraph from the docstring
+                    function_description = function.__doc__.strip().split("\n\n")[0]
+                # Last resort - use function name
+                else:
+                    function_description = f"Function {function_name}"
+
+                # Create a dynamic wrapper function that exposes the actual parameters
+                def create_function_wrapper(func: Function, schema: type[BaseModel]):
+                    # Check if we're dealing with AIQChatRequest - special case
+                    from inspect import Parameter, Signature
+
+                    is_chat_request = False
+
+                    # Check if the schema name is AIQChatRequest
+                    if schema.__name__ == "AIQChatRequest" or (
+                        hasattr(schema, "__qualname__") and "AIQChatRequest" in schema.__qualname__
+                    ):
+                        is_chat_request = True
+                        logger.info(f"Function {function_name} uses AIQChatRequest - creating simplified interface")
+
+                        # For AIQChatRequest, we'll create a simple wrapper with just a query parameter
+                        parameters = [
+                            Parameter(
+                                name="query", kind=Parameter.KEYWORD_ONLY, default=Parameter.empty, annotation=str
+                            )
+                        ]
+
+                        field_info = {
+                            "query": {"type": str, "description": "Input query or prompt text", "required": True}
+                        }
+                    else:
+                        # Regular case - extract parameter information from the input schema
+                        # Extract parameter information from the input schema
+                        param_fields = schema.model_fields
+
+                        from pydantic import create_model
+                        import inspect
+                        import re
+
+                        # Parse docstring if available to get parameter descriptions
+                        param_descriptions = {}
+                        if func.__doc__:
+                            docstring = func.__doc__
```

Comment:
I don't think we need to use docstring any more. We moved to field.annotation to unify the `aiq info components` display.